Mathlib/RingTheory/SimpleRing/TensorProduct.lean

```diff
@@ -0,0 +1,362 @@
+/-
+Copyright (c) 2025 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Yunzhou Xie
+-/
+import Mathlib.Algebra.Central.Basic
+import Mathlib.LinearAlgebra.Basis.VectorSpace
+import Mathlib.RingTheory.Flat.Basic
+import Mathlib.RingTheory.TwoSidedIdeal.SpanAsSum
+
+/-!
+# Tensor product of simple algebras
+
+In this file, we show that the tensor product between a simple algebra and a central simple algebra
+is simple. This a pre-requist for proving the group law of the Brauer group, in particular it helps
+show that tensor product of two central simple algebras is central simple.
+
+## Main Results
+
+* `TensorProduct.nontrivial`: The tensor product of two non-trivial algebras is non-trivial.
+* `TensorProduct.simple`: The tensor product of a simple algebra and a central simple algebra
+  is simple.
+
+## References
+
+* [StackProject 074B](https://stacks.math.columbia.edu/tag/074B)
+
+## Tags
+Noncommutative algebra, tensor product, simple algebra, central simple algebra
+
+-/
+
+variable (K : Type*) [Field K]
+
+universe u v
+
+open scoped TensorProduct
+
+variable {K} in
+/--
+a non-zero element in an ideal that can be represented as a sum of tensor products of `n`-terms.
+-/
+structure is_obtainable_by_sum_tmul
+    {ιA A B : Type*} [Ring A] [Algebra K A] [Ring B] [Algebra K B]
+    (x : A ⊗[K] B) (𝒜 : Basis ιA K A) (I : TwoSidedIdeal (A ⊗[K] B)) (n : ℕ) : Prop where
+  mem : x ∈ I
+  ne_zero : x ≠ 0
+  rep : ∃ (s : Finset ιA) (_ : s.card = n) (f : ιA → B),
+    x = ∑ i ∈ s, 𝒜 i ⊗ₜ[K] f i
+
+variable {K} in
+lemma is_obtainable_by_sum_tmul.exists_minimal_element
+    {A B : Type*} [Ring A] [Algebra K A] [Ring B] [Algebra K B]
+    (ιA : Type*) (𝒜 : Basis ιA K A)
+    (I : TwoSidedIdeal (A ⊗[K] B)) (hI : I ≠ ⊥) :
+    ∃ (n : ℕ) (x : A ⊗[K] B), is_obtainable_by_sum_tmul x 𝒜 I n ∧
+      ∀ (m : ℕ) (y : A ⊗[K] B) , is_obtainable_by_sum_tmul y 𝒜 I m → n ≤ m := by
```

Comment:
I know that [Stacks 074B](https://stacks.math.columbia.edu/tag/074B) explicitly uses the language of "non-zero" and "minimal", but that is maths speak. In Lean, the translation is to ignore non-zero, and instead of "minimal" you do induction.